src/uint/modular/runtime_mod/runtime_pow.rs

```diff
@@ -35,8 +39,72 @@ impl<const LIMBS: usize> DynResidue<LIMBS> {
     }
 }
 
-impl<const LIMBS: usize> PowBoundedExp<Uint<LIMBS>> for DynResidue<LIMBS> {
-    fn pow_bounded_exp(&self, exponent: &Uint<LIMBS>, exponent_bits: usize) -> Self {
+impl<const LIMBS: usize, const RHS_LIMBS: usize> PowBoundedExp<Uint<RHS_LIMBS>>
+    for DynResidue<LIMBS>
+{
+    fn pow_bounded_exp(&self, exponent: &Uint<RHS_LIMBS>, exponent_bits: usize) -> Self {
         self.pow_bounded_exp(exponent, exponent_bits)
     }
 }
+
+impl<const N: usize, const LIMBS: usize, const RHS_LIMBS: usize>
+    MultiExponentiateBoundedExp<Uint<RHS_LIMBS>, [(Self, Uint<RHS_LIMBS>); N]>
+    for DynResidue<LIMBS>
+{
+    fn multi_exponentiate_bounded_exp(
+        bases_and_exponents: &[(Self, Uint<RHS_LIMBS>); N],
+        exponent_bits: usize,
+    ) -> Self {
+        // TODO: is it fine? will crash if N == 0
```

Comment:
```suggestion
```
@tarcieri can I document that the function panics if the slice is empty?

Reply:
It feels like the sort of thing where it should be possible to do some kind of const assertion, although I forget how to work around the "can't use generic parameters from outer item" problem

Reply:
Posted a solution [here](https://github.com/RustCrypto/crypto-bigint/pull/248#discussion_r1397661028)